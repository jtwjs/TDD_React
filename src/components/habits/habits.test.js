import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import Habits from "./habits";
import userEvent from "@testing-library/user-event";

describe("Habits", () => {
  const habits = [
    { name: "taewoong", id: 1, count: 1 },
    { name: "woong", id: 2, count: 2 },
  ];

  let onReset = jest.fn();

  const HabitComponent = (
    <Habits
      habits={habits}
      onIncrement={jest.fn()}
      onDecrement={jest.fn()}
      onDelete={jest.fn()}
      onAdd={jest.fn()}
      onReset={onReset}
    />
  );
  it("renders", () => {
    const component = renderer.create(HabitComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Button Click", () => {
    beforeEach(() => {
      render(HabitComponent);
    });

    it("calls onReset when clicking 'reset' button", () => {
      const resetBtn = screen.getByText("Reset All");
      userEvent.click(resetBtn);
      expect(onReset).toHaveBeenCalledTimes(1);
    });
  });
});
