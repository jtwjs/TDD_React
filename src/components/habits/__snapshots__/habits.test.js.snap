// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits renders 1`] = `
<div
  className="habits"
>
  <form
    className="add-form"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li
      className="habit"
    >
      <span
        className="habit-name"
        data-testid="name"
      >
        taewoong
      </span>
      <span
        className="habit-count"
        data-testid="count"
      >
        1
      </span>
      <button
        aria-label="increase"
        className="habit-button habit-increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        aria-label="decrease"
        className="habit-button habit-decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        aria-label="delete"
        className="habit-button habit-delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
        data-testid="name"
      >
        woong
      </span>
      <span
        className="habit-count"
        data-testid="count"
      >
        2
      </span>
      <button
        aria-label="increase"
        className="habit-button habit-increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        aria-label="decrease"
        className="habit-button habit-decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        aria-label="delete"
        className="habit-button habit-delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
  <button
    className="habits-reset"
    onClick={[MockFunction]}
  >
    Reset All
  </button>
</div>
`;
